client/verta/verta/_tracking/experimentrun.py

```diff
@@ -1239,14 +1239,15 @@ def log_model(self, model, custom_modules=None, model_api=None, artifacts=None,
         except (TypeError, ValueError):
             extension = None
         _utils.THREAD_LOCALS.active_experiment_run = self
-        if isinstance(model, six.string_types):  # filepath
-            model = open(model, 'rb')
+        if isinstance(model, six.string_types):
+            if os.path.isdir(model):
+                model = _artifact_utils.zip_dir(model)
+            else:  # filepath
+                model = open(model, 'rb')
```

Comment:
Here and in other places in this module I think we're neglecting to close the file object. Supposedly it's something which some Python implementations handle automatically, but the behavior might be implementation specific.

Reply:
Yes! There are quite a few places where file objects are created, and not explicitly closed. It's potetially problematic for everything other than CPython.

Fixing it is complicated by the fact that some of these methods, e.g. for artifact upload, were written a bit naïvely by as less-experienced me to accommodate file objects passed in by the user. I wanted to avoid closing a caller's file object, but didn't have a good way to distinguish when it was one created internally (since they're all bound to the same vars), so ended up not closing any of them.

Reply:
I think it's worth a follow-up, but I'm not sure that I can resolve it even just within the scope of this ticket.